python/src/lib.rs

```diff
@@ -1127,6 +1128,71 @@ impl From<&PyAddAction> for Add {
     }
 }
 
+#[pyfunction]
+#[allow(clippy::too_many_arguments)]
+fn write_to_deltalake(
+    table_uri: String,
+    data: PyArrowType<ArrowArrayStreamReader>,
+    // schema: Option<PyArrowType<ArrowSchema>>, // maybe do the schema casting on python side
+    mode: String,
+    max_rows_per_group: i64,
+    overwrite_schema: bool,
+    partition_by: Option<Vec<String>>,
+    name: Option<String>,
+    description: Option<String>,
+    configuration: Option<HashMap<String, Option<String>>>,
+    storage_options: Option<HashMap<String, String>>,
+) -> PyResult<()> {
+    let batches = data.0.map(|batch| batch.unwrap()).collect::<Vec<_>>();
+    let save_mode = save_mode_from_str(&mode)?;
+
+    let mut metadata: HashMap<String, Value> = HashMap::new();
+
+    if let Some(name) = name {
+        metadata.insert("name".to_string(), name.into());
+    }
+
+    if let Some(description) = description {
+        metadata.insert("description".to_string(), description.into());
+    }
+
+    if let Some(configuration) = configuration {
+        metadata.insert("configuration".to_string(), json!(configuration));
+    }
+
+    // // This should be done when the table can not be loaded ...
+    // match save_mode {
+    //     SaveMode::Ignore => {
+    //         return Ok(())
+    //     }
+    //     _ => ()
+    // }
+    let options = storage_options.clone().unwrap_or_default();
+    let table = rt()?
+        .block_on(DeltaOps::try_from_uri_with_storage_options(
+            &table_uri, options,
+        ))
+        .map_err(PythonError::from)?;
+
+    let mut builder = table
+        .write(batches)
+        .with_save_mode(save_mode)
+        .with_overwrite_schema(overwrite_schema)
+        .with_metadata(metadata)
```

Comment:
This name is a little bit misleading. It does not refer to the metadata action, but rather additional application-specific metadata. Took a quick look at the code, but it seems table configuration it not passed through to the create operation in the "create if not exists" path.

__update__: just saw you mentioned that already :).